src/main/java/com/google/firebase/remoteconfig/ListVersionsOptions.java

```diff
@@ -0,0 +1,205 @@
+    private String pageToken;
+    private String endVersionNumber;
+    private String startTime;
+    private String endTime;
+
+    private Builder() {}
+
+    private Builder(ListVersionsOptions options) {
+      this.pageSize = options.pageSize;
+      this.pageToken = options.pageToken;
+      this.endVersionNumber = options.endVersionNumber;
+      this.startTime = options.startTime;
+      this.endTime = options.endTime;
+    }
+
+    /**
+     * Sets the page size.
+     *
+     * @param pageSize The maximum number of items to return per page.
+     * @return This builder.
+     */
+    public Builder setPageSize(int pageSize) {
+      this.pageSize = pageSize;
+      return this;
+    }
+
+    /**
+     * Sets the page token.
+     *
+     * @param pageToken The {@code nextPageToken} value returned from a previous List request,
+     *                  if any.
+     * @return This builder.
+     */
+    public Builder setPageToken(String pageToken) {
+      this.pageToken = pageToken;
+      return this;
+    }
+
+    /**
+     * Sets the newest version number to include in the results.
+     *
+     * @param endVersionNumber Specify the newest version number to include in the results.
+     *                         If specified, must be greater than zero. Defaults to the newest
+     *                         version.
+     * @return This builder.
+     */
+    public Builder setEndVersionNumber(String endVersionNumber) {
+      this.endVersionNumber = endVersionNumber;
+      return this;
+    }
+
+    /**
+     * Sets the newest version number to include in the results.
+     *
+     * @param endVersionNumber Specify the newest version number to include in the results.
+     *                         If specified, must be greater than zero. Defaults to the newest
+     *                         version.
+     * @return This builder.
+     */
+    public Builder setEndVersionNumber(long endVersionNumber) {
+      this.endVersionNumber = String.valueOf(endVersionNumber);;
+      return this;
+    }
+
+    /**
+     * Sets the earliest update time to include in the results.
+     *
+     * @param startTimeMillis Specify the earliest update time to include in the results.
+     *                        Any entries updated before this time are omitted.
+     * @return This builder.
+     */
+    public Builder setStartTimeMillis(long startTimeMillis) {
+      this.startTime = convertToUtcZuluFormat(startTimeMillis);
+      return this;
+    }
+
+    /**
+     * Sets the latest update time to include in the results.
+     *
+     * @param endTimeMillis Specify the latest update time to include in the results.
+     *                      Any entries updated on or after this time are omitted.
+     * @return This builder.
+     */
+    public Builder setEndTimeMillis(long endTimeMillis) {
+      this.endTime = convertToUtcZuluFormat(endTimeMillis);
+      return this;
+    }
+
+    private String convertToUtcZuluFormat(long millis) {
```

Comment:
Probably belongs in the shared util. I assume there are other places where this is needed.

Reply:
Good call. Currently, we use it only in `ListVersionsOptions`. I moved it to Utils anyway to future proof :)